otel/propagator.go

```diff
@@ -0,0 +1,137 @@
+//go:build go1.18
+
+package sentryotel
+
+import (
+	"context"
+
+	"github.com/getsentry/sentry-go"
+	"go.opentelemetry.io/otel/baggage"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type sentryPropagator struct{}
+
+func NewSentryPropagator() propagation.TextMapPropagator {
+	return &sentryPropagator{}
+}
+
+// Inject sets Sentry-related values from the Context into the carrier.
+//
+// https://opentelemetry.io/docs/reference/specification/context/api-propagators/#inject
+func (p sentryPropagator) Inject(ctx context.Context, carrier propagation.TextMapCarrier) {
+	sentry.Logger.Printf("\n--- Propagator Inject\nContext: %#v\nCarrier: %#v\n", ctx, carrier)
```

Comment:
This might be super noisy for users. I use `Debug: true` a lot, so maybe we could reduce the output a little bit.